### ISSUE TITLE
Visualizer Installation Documentation is Inaccurate

### ISSUE THREAD

**Issue by jamesbascle** [NRules/NRules#347]
When running `build.bat` and `build.bat Build Visualizer` the expected folder, `<Repo Root>\binaries\NRules.Debugger.Visualizer` does not seem to show up.  There are binaries for .NET 4.7.2 and .NET Standard 2.0 in the `<Repo Root>\src\NRules.Debugger.Visualizer\NRules.Debugger.Visualizer\bin\Release\net472` and `<Repo Root>\NRules.Debugger.Visualizer\NRules.Debugger.Visualizer.DebuggeeSide\bin\Release\netstandard2.0` folders respectively.

When I copy the files from the former to `<Documents>\Visual Studio <Version>\Visualizers` I get the following error upon clicking the View button on a Session.
![image](https://github.com/NRules/NRules/assets/4188293/412b32d3-9de3-4e78-9bcd-10982f70d6da)

When I copy the files from the latter to `<Documents>\Visual Studio 2022\Visualizers\netstandard2.0` I get the following error upon clicking the View button on a Session. 
![image](https://github.com/NRules/NRules/assets/4188293/c4d13103-ad47-4f72-98a7-88867139f8bc)

I am on latest Visual Studio 2022 with the DGML Editor installed.

**Comment by snikolayev** [comment]
@jamesbascle no worries, I can help you figure it out. 

I assume you checked out the latest code from main or develop branch from GitHub. If this is not the case and you are looking at some old version, LMK.

This is what the folder structure should be under the `Documents\Visual Studio 2022`:
```
- Visualizers
  |
  - NRules.Debugger.Visualizer.dll
  - netstandard2.0
    |
    - NRules.Debugger.Visualizer.DebuggeeSide.dll
```
You can try manually arranging the files into the folder structure like that, but really, the build script should have created it for you under the `build\bin\NRules.Debugger.Visualizer`.
And as I was typing it, I see that the docs were pointing to an old location. I just fixed the documentation, so please see if this fixes it for you.

Also, please make sure that the version number in the `build.ps` matches the version of the NRules you are using. The one in develop branch has been incremented, so you may want to update it to match the one you are testing with. Visual Studio is very picky about the version matching exactly.

**Comment by jamesbascle** [comment]
Thanks for the reply Sergiy!

I just wiped out and re-cloned the repo.  I checked out commit tagged 0.9.4, d7f783904d2771e1ccd79b3fc34914059a991ed5, which is the version of NRules and NRules.Fluent I have installed.  build.ps1 has 0.9.4 in it as expected.

After running the build commands again, When I copy `\NRules\build\bin\NRules.Debugger.Visualizer` to `Documents\Visual Studio 2022\Visualizers` I get the second error modal again upon clicking View.

**Comment by jamesbascle** [comment]
I've done the above procedure but also having restarted my computer.  So:

- restart
- delete repo
- reclone
- update to commit
- build nrules
- build visualizer
- copy from `build\bin\NRules.Debugger.Visualizer` to `Documents\Visual Studio 2022\Visualizers`
- open my solution in VS
- run test and set breakpoint after running my session
- click `View` in the debugger
-  see 
![image](https://github.com/NRules/NRules/assets/4188293/a8fab562-8c97-450e-b1ce-00e1401948d1)

It may be some other configuration issue with my system, but I can't imagine what.  I  just set up this system about a month or so ago and only have a handful of dev tools, Firefox, Docker, and Teams installed.  What is strange is that on a previous version I actually did get it set up and get the DGML generated, but VS was opening it up in another visual studio session and showed me the raw markup but did not do the visualizer.  This was before i updated to 0.9.3 and then 0.9.4.  I don't even have the code that yielded that experience anymore since it was a throwaway experiment, so I'm not sure what version it was on.

I'm going to try to get a couple of co-workers to give it a shot tomorrow to see if we can reproduce my issue on their machines.  If you can think of anything else I'll give it a shot as well.

Again, thanks for the reply!

**Comment by snikolayev** [comment]
This is indeed really weird. I just checked out from the tag, built the visualizer and copied to the same location. Created a simple console app with one simple rule, targeting .net 8, installed NRules 0.9.4 from nuget.org. Stopped in debugger and clicked the magnifying glass on the session object, and it all worked just fine.
One thing you can double-check is that there are no left-overs from some older versions of this visualizer under `C:\Program Files\Microsoft Visual Studio\2022\<Edition>\Common7\Packages\Debugger\Visualizers`

**Comment by snikolayev** [comment]
I'm moving this into discussions

**Comment by snikolayev** [comment]
@jamesbascle I have another idea. The way the visualizer works, it actually generates a DGML file under %Temp% folder, and opens it via shell. The error `Application not found` may be suggesting that it just can't find any application associated with the .dgml extension.
Can you please go to your %Temp% folder and look for the `session.dgml` file? If it's there, try opening it, and see if it opens in VS or if Windows complains there is no associated program with it. Delete the file, repeat the test, and when you see the error check if that file got re-created.

**Comment by jamesbascle** [comment]
Okay, we're onto something here!  I set the program for that session.dgml file to VS, and now when i hit inspect it pops open another VS2022 instance and shows me the graph!  I would have expected it to open the file inside of the same instance I'm debugging inside of.

**Comment by snikolayev** [comment]
Glad to hear we got to the bottom of it. Interestingly, it does open it in the same VS instance for me, so I think there is still something messed up with the file association on your computer. One thing I can think of is if you have multiple VS versions installed, and it's opening up via the VS version selector. 
I would also try rebooting after you've added the file association and trying again.
Another thing you can try is removing the file association (google it, you can do via regedit), then uninstall the DGML feature in VS and re-install it. Maybe it adds it for you in a proper way.

**Comment by jamesbascle** [comment]
Okay I deleted *every* key in my registry named `.dgml`, there were 4 or 5 of them scattered across various hierarchies.  I then uninstalled the dgml editor, reinstalled it, and now the association looks like a lil graph rather than VS2022's icon.  

It's now behaving as expected!  No idea how my .dgml association got so out of wack, but I'm glad it's fixed now!

Thanks a ton for your help and idea on this Sergiy!
